---
title: "Class13: Genome informatics and high throughput sequencing (Part 1)"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```



# MXL genotype analysis 

####How many GIG genotype are there in the 1000 genome data for hte MXL group?

#####I obtained the input from:
http://uswest.ensembl.org/Homo_sapiens/Variation/Sample?db=core;r=17:39894595-39895595;v=rs8067378;vdb=variation;vf=362656008#373531_tablePanel

```{r}
mxl<-read.csv("373531-SampleGenotypes-Homo_sapiens_Variation_Sample_rs8067378.csv")
head(mxl)
```

```{r}
#here we used table to differentiate the different genotypes 
table(mxl$Genotype..forward.strand.)
```

```{r}
round(table(mxl$Genotype..forward.strand.)/nrow(mxl)*100,2)
```


```{r}
#install.packages("seqinr")
#install.packages("gtools")

library(seqinr)
library(gtools)
phred <- asc( s2c("DDDDCDEDCDDDDBBDDDCC@") ) - 33 
phred


```

##genotype expression 
```{r}
expr<-read.table("rs8067378_ENSG00000172057.6.txt")
head(expr)

```


```{r}
table(expr$geno)
```


```{r}
inds.gg<-expr$geno=="G/G"
summary(expr[inds.gg,"exp"])

```

```{r}
inds.ag<-expr$geno=="A/G"
summary(expr[inds.gg,"exp"])
```


```{r}
inds.gg<-expr$geno=="A/A"
summary(expr[inds.gg,"exp"])
```

```{r}
boxplot(exp~geno,data=expr,notch=TRUE)
```

```{r}
library(ggplot2)
ggplot(expr, aes(geno, exp, fill=geno)) + 
  geom_boxplot(notch=TRUE, outlier.shape = NA) + 
  geom_jitter(shape=16, position=position_jitter(0.2), alpha=0.4)
```



